h2o-py/tests/testdir_misc/pyunit_ice_plot.py

```diff
@@ -81,20 +84,20 @@ def test_handle_orig_values():
                 orig_value_prediction = _handle_orig_values(is_factor, tmp, encoded_col, plt, target, gbm,
                                                             frame, index, column, colors[i], percentile_string)
 
+                if (is_factor and math.isnan(factor_map([frame[index, column]])[0])) or (not is_factor and math.isnan(frame[index, column])):
+                    orig_test_value = orig_value_prediction["mean_response"][orig_value_prediction.nrow - 1]
```

Comment:
I'm not sure if I understand this part of code - is `orig_test_value` supposed to be the last value from the `orig_value_prediction["mean_response"]`? If so, consider the following suggestion.
```suggestion
                    orig_test_value = orig_value_prediction["mean_response"][-1]
```

Also, I am not sure I understand the reason why for factors we take the last row (`orig_value_prediction.nrow -1`) and for the rest we don't subset. I'd understand if we'd take the last column since the predictions for classification are in a form of a vector of probabilities + predicted class. Could you shed some light on that?

Reply:
for nan's we take last value, regardless of whether it is a factor or not.

If we calculate new original value, _handle_orig_values returns new row which is then added to the overall data with other points. if original value is nan, its value was already calculated during 
```
tmp = NumpyFrame(
                model.partial_plot(
                    frame,
                    cols=[column],
                    plot=False,
                    row_index=index,
                    targets=target,
                    nbins=100 if not is_factor else 1 + frame[column].nlevels()[0],
                    include_na=True
                )[0]
            )
```

thanks to include_na=True and we don't need to append anything to tmp:

```
orig_row = _handle_orig_values(is_factor, tmp, encoded_col, plt, target, model,
                                frame, index, column, colors[i], percentile_string)
            if not _isnan(frame.as_data_frame()[column][index]):
                tmp._data=np.append(tmp._data, orig_row._data, axis=0)
```

 and _handle_orig_values in this case returns whole just to be able to test it as a whole for all cases. Another possibility is, to not to return anything for nan case and test this value directly from tmp. WDYT?

Reply:
Thank you! It makes sense to me now.